test/dataset/test_collation.py

```diff
@@ -0,0 +1,27 @@
+from lhotse.dataset.collation import TokenCollater
+from lhotse import CutSet
+from lhotse.testing.dummies import dummy_cut, dummy_supervision
+
+
+def test_token_collater():
+    test_sentences = [
+        "Testing the first sentence.",
+        "Let's add some more punctuation, shall we?",
+        "How about number 42!"
+    ]
+
+    cuts = CutSet.from_cuts(
+        dummy_cut(
+            idx,
+            idx,
+            supervisions=[dummy_supervision(
+                idx, idx, text=sentence
+            )]
+        ) for idx, sentence in enumerate(test_sentences)
+    )
+
+    token_collater = TokenCollater(cuts)
+    tokens_batch, tokens_lens = token_collater(cuts)
```

Comment:
would be good to assert some properties of tokens_batch and tokens_lens too (that they are int tensors, have the right shapes, and maybe specific values for lens)

Reply:
adding